src/figuretype/depot.c

```diff
@@ -0,0 +1,395 @@
+            is_linked = 1;
+        }
+    }
+    if (!is_linked) {
+        f->state = FIGURE_STATE_DEAD;
+        update_image(f);
+        return;
+    }
+
+    switch (f->action_state) {
+        case FIGURE_ACTION_150_ATTACK:
+            figure_combat_handle_attack(f);
+            break;
+        case FIGURE_ACTION_149_CORPSE:
+            figure_combat_handle_corpse(f);
+            break;
+        case FIGURE_ACTION_238_DEPOT_CART_PUSHER_INITIAL:
+            set_cart_graphic(f);
+            if (!map_routing_citizen_is_passable(f->grid_offset)) {
+                f->state = FIGURE_STATE_DEAD;
+            }
+
+            if (is_order_condition_satisfied(b, &b->data.depot.current_order)) {
+                building *src = building_get(b->data.depot.current_order.src_storage_id);
+                map_point road_access;
+                get_storage_road_access(src, &road_access);
+                f->action_state = FIGURE_ACTION_239_DEPOT_CART_PUSHER_HEADING_TO_SOURCE;
+                f->wait_ticks = DEPOT_CART_REROUTE_DELAY + 1;
+                f->destination_building_id = b->data.depot.current_order.src_storage_id;
+                f->destination_x = road_access.x;
+                f->destination_y = road_access.y;
+            } else {
+                f->state = FIGURE_STATE_DEAD;
+            }
+
+            f->image_offset = 0;
+            break;
+        case FIGURE_ACTION_239_DEPOT_CART_PUSHER_HEADING_TO_SOURCE:
+            set_cart_graphic(f);
+            if (f->wait_ticks > DEPOT_CART_REROUTE_DELAY) {
+                figure_movement_move_ticks_with_percentage(f, speed_factor, percentage_speed);
+                if (f->direction == DIR_FIGURE_AT_DESTINATION) {
+                    f->action_state = FIGURE_ACTION_240_DEPOT_CART_PUSHER_AT_SOURCE;
+                    f->wait_ticks = 0;
+                } else if (f->direction == DIR_FIGURE_LOST) {
+                    f->action_state = FIGURE_ACTION_244_DEPOT_CART_PUSHER_CANCEL_ORDER;
+                    f->wait_ticks = 0;
+                } else if (f->direction == DIR_FIGURE_REROUTE) {
+                    figure_route_remove(f);
+                    f->wait_ticks = 0;
+                }
+            } else {
+                f->wait_ticks++;
+            }
+            break;
+        case FIGURE_ACTION_240_DEPOT_CART_PUSHER_AT_SOURCE:
+            set_cart_graphic(f);
+            f->wait_ticks++;
+            if (f->wait_ticks > DEPOT_CART_LOAD_OFFLOAD_DELAY) {
+                building *src = building_get(b->data.depot.current_order.src_storage_id);
+
+                // TODO upgradable?
+                int capacity = resource_is_food(b->data.depot.current_order.resource_type) ? DEPOT_CART_PUSHER_FOOD_CAPACITY : DEPOT_CART_PUSHER_OTHER_CAPACITY;
+                int remaining_capacity = storage_remove_resource(src, b->data.depot.current_order.resource_type, capacity);
+                if (remaining_capacity == capacity) {
```

Comment:
I never liked these empty `if` clauses. Just invert the condition and remove the empty clause please.

(disclaimer: I know this code isn't originally yours)

Reply:
Done.